Userland/Libraries/LibJS/Bytecode/Interpreter.cpp

```diff
@@ -1562,6 +1562,14 @@ ThrowCompletionOr<void> DeleteByIdWithThis::execute_impl(Bytecode::Interpreter&
 
 ThrowCompletionOr<void> ResolveThisBinding::execute_impl(Bytecode::Interpreter& interpreter) const
 {
+    auto& running_execution_context = interpreter.vm().running_execution_context();
+    if (running_execution_context.function && is<ECMAScriptFunctionObject>(*running_execution_context.function)) {
+        auto const& esfo = static_cast<ECMAScriptFunctionObject&>(*running_execution_context.function);
+        if (!esfo.allocates_function_environment()) {
+            interpreter.set(dst(), running_execution_context.this_value);
+            return {};
+        }
+    }
```

Comment:
This should moved into the `if (cached_this_value.is_empty()) {` block below! Should make it even faster :)